### PR TITLE
Added pre post cb functions for erpc calls.

### PR DESCRIPTION
**Pull request by Hadatko**
Also with default implementation for FreeRTOS.

Related to #130

### CODE REVIEW

**Comment by Hadatko** [comment]
@MichalPrincNXP please do not merge this one until we agree on it. I need test the code. It should be fine (partially tested). 
So what is your opinion about that?

**Comment by MichalPrincNXP** [comment]
> 
> 
> @MichalPrincNXP please do not merge this one until we agree on it. I need test the code. It should be fine (partially tested).
> So what is your opinion about that?

Understand. So far it seems to be OK. I think having this possibility in generated shim code is better than having it in individual transports - more general, less complicated when creating a new transport. The only thing I have noticed is missing python support. Thanks.

File: erpc_c/port/erpc_setup_extensions_freertos.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+void erpc_call_timer_cb_default(TimerHandle_t xTimer)
+{
+    assert(1 != 1 && "eRPC task freezed.");
+}
+
+void erpc_create_timer_default(uint32_t waitTime = 5 * 60 * 1000)
+{
+    g_erpc_call_timer_cb = xTimerCreate("Erpc client call timer", waitTime, pdFALSE, NULL, erpc_call_timer_cb_default);
```
Comment (qcabrol) [inline erpc_c/port/erpc_setup_extensions_freertos.cpp:39]: for Freertos context, having conversion from ms to Ticks would ensure coherent timing of that default timer on different platforms. (use freeRTOS related define value)

File: erpc_c/port/erpc_setup_extensions_freertos.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    assert(g_erpc_call_timer_cb && "Creating eRPC timer failed.");
+}
+
+bool is_erpc_call_executed_default()
```
Comment (qcabrol) [inline erpc_c/port/erpc_setup_extensions_freertos.cpp:43]: Routine name makes it a bit confusing

**Review by qcabrol — COMMENTED** [review]
Quite happy about the additional abstraction offered by the pre/post callbacks ! Just added 2 little comments on my side. Thanks @Hadatko

**Comment by Hadatko** [comment]
> > @MichalPrincNXP please do not merge this one until we agree on it. I need test the code. It should be fine (partially tested).
> > So what is your opinion about that?
> 
> Understand. So far it seems to be OK. I think having this possibility in generated shim code is better than having it in individual transports - more general, less complicated when creating a new transport. The only thing I have noticed is missing python support. Thanks.

Yes there is missing python support. Client is independed to server. So in combination C<->Python it can be used on C side. I think if somebody will need the same on python side it should be another PR. I can imagine if eRPC will be written for another language like nodejs it would be hard to add for programmer same feature to all different programm languages. I also understand that add features to only one programm language may lead to some divergent eRPC versions. On another hand all features may not be requested for all programm languages...

What do you think? Can be this potentionaly merged without python support? If this feature will be documented somewhere we can notice that currently only C is supported.

File: erpc_c/port/erpc_setup_extensions_freertos.cpp (hunk 1 above)
Comment (Hadatko) [reply erpc_c/port/erpc_setup_extensions_freertos.cpp:39]: Correct, i will do that.

File: erpc_c/port/erpc_setup_extensions_freertos.cpp (hunk 2 above)
Comment (Hadatko) [reply erpc_c/port/erpc_setup_extensions_freertos.cpp:43]: What do you suggest? It is about to inform user that erpc call is executed, so task could potentionaly freezed, or the execution is still in progress.

**Review by MichalPrincNXP — APPROVED** [review]

**Comment by MichalPrincNXP** [comment]
Thank you, Dusan!

**Pull request merged**